File: leetcode/array/leetcode_35.java
package leetcode.array;

public class leetcode_35 {
    public static void main(String[] args) {
        leetcode_35 lc = new leetcode_35();

        int[] arr = {1, 3, 5, 6};
        System.out.println(lc.searchInsert(arr, 5));
    }

    public int searchInsert(int[] nums, int target) {
        int low = 0, high = nums.length;

        while (low < high) {
            int mid = low + (high - low) / 2;
            if (nums[mid] >= target) high = mid;
            else low = mid + 1;
        }

        return low;
    }
}

File: search/LeetCode_35.kt
package search

class LeetCode_35 {

    fun searchInsert(nums: IntArray, target: Int): Int {
        var low = 0
        var high = nums.lastIndex

        while (low <= high) {
            val middle = (low + high) / 2

            if (nums[middle] == target) {
                return middle
            } else if (nums[middle] > target) {
                high = middle - 1
            } else {
                low = middle + 1
            }
        }

        return low
    }
}

fun main() {
    // Input
    val nums = intArrayOf(1, 3, 5, 6)
    val target = 5
    // Output
    LeetCode_35().searchInsert(nums, target).let {
        println(it)
        require(it == 2)
    }
}
